contracts/ExampleToken.cdc

```diff
@@ -44,228 +81,143 @@ pub contract ExampleToken: FungibleToken {
-                        createEmptyVaultFunction: (fun (): @ExampleToken.Vault {
-                            return <-ExampleToken.createEmptyVault()
-                        })
-                    )
-                case Type<FungibleTokenMetadataViews.TotalSupply>():
-                    return FungibleTokenMetadataViews.TotalSupply(totalSupply: ExampleToken.totalSupply)
-            }
-            return nil
+        access(all) fun createEmptyVault(): @ExampleToken.Vault {
+            return <-create Vault(balance: 0.0)
         }
     }
 
-    /// Function that creates a new Vault with a balance of zero
-    /// and returns it to the calling context. A user must call this function
-    /// and store the returned Vault in their storage in order to allow their
-    /// account to be able to receive deposits of this token type.
-    ///
-    /// @return The new Vault resource
+    /// Minter
     ///
-    pub fun createEmptyVault(): @Vault {
-        return <-create Vault(balance: 0.0)
-    }
-
-    pub resource Administrator {
-
-        /// Function that creates and returns a new minter resource
-        ///
-        /// @param allowedAmount: The maximum quantity of tokens that the minter could create
-        /// @return The Minter resource that would allow to mint tokens
-        ///
-        pub fun createNewMinter(allowedAmount: UFix64): @Minter {
-            emit MinterCreated(allowedAmount: allowedAmount)
-            return <-create Minter(allowedAmount: allowedAmount)
-        }
-
-        /// Function that creates and returns a new burner resource
-        ///
-        /// @return The Burner resource
-        ///
-        pub fun createNewBurner(): @Burner {
-            emit BurnerCreated()
-            return <-create Burner()
-        }
-    }
-
     /// Resource object that token admin accounts can hold to mint new tokens.
     ///
-    pub resource Minter {
-
-        /// The amount of tokens that the minter is allowed to mint
-        pub var allowedAmount: UFix64
-
+    access(all) resource Minter {
+        /// mintTokens
+        ///
         /// Function that mints new tokens, adds them to the total supply,
         /// and returns them to the calling context.
         ///
-        /// @param amount: The quantity of tokens to mint
-        /// @return The Vault resource containing the minted tokens
-        ///
-        pub fun mintTokens(amount: UFix64): @ExampleToken.Vault {
-            pre {
-                amount > 0.0: "Amount minted must be greater than zero"
-                amount <= self.allowedAmount: "Amount minted must be less than the allowed amount"
-            }
+        access(all) fun mintTokens(amount: UFix64): @ExampleToken.Vault {
             ExampleToken.totalSupply = ExampleToken.totalSupply + amount
-            self.allowedAmount = self.allowedAmount - amount
-            emit TokensMinted(amount: amount)
+            emit TokensMinted(amount: amount, type: self.getType().identifier)
```

Comment:
Do we need to include type in the event if `ExampleToken` only defines one token in the contract? Was thinking this is not a standard-level event so can be custom for the implementation

Reply:
We don't have to, but it is still a good practice to, especially since this is an example that others might follow if they are defining multiple NFT types